xapian-letor/tests/api_letor.cc

```diff
@@ -94,26 +167,169 @@ DEFINE_TESTCASE(createfeaturevector, generated)
     TEST(fv.empty());
```

Comment:
> Just before you run FeatureList::create_feature_vectors, you can check for an empty MSet. This will indicate that an empty feature vector is created for an empty MSet

Not sure I follow - are you saying we should check `Xapian::MSet mset;` actually gives an empty `MSet`?

Such a check primary belongs in the xapian-core testsuite (I actually don't see any test of exactly that there currently).  There's an argument for testing things you rely on from the layers you're building on, and `TEST(mset.empty());` is a succinct way to document exactly what the testcase is relying on.  In more complex cases it would certainly be reasonable, and also if we've seen the feature in question work incorrectly!  "A default constructed `Xapian::MSet` is empty" seems both obvious and very likely to work correctly, but I wouldn't object to such a check even here.

> Also I think that adding this test condition only here is enough. You need not test the creation of an empty FeatureVector by running FeatureList::create_feature_vectors on empty MSet in the remaining tests as it would be redundant. Although I think @ojwb might better advise in this regard.

I think I agree - unless I'm missing some detail, this particular part of this testcase is testing the exact same thing as the equivalent part of the "tigers" testcase AyushP123 noted below.  Improving test coverage is the main goal, not making the tests take longer to run.

Reply:
BTW, you can use C++11 `auto` here to avoid having to write out the long type name and avoid having to wrap the line:

```
    auto fv = fl.create_feature_vectors(mset, Xapian::Query("lions"), db);
```

Reply:
The singular "tigers" case was a case in which the code is tested for a single document, with a data-rich environment, looking over a case for a single file in the db might be a common mistake.
So I just added one for anyone overlooking that.